AnnoDesigner/App.xaml.cs

```diff
@@ -221,16 +221,20 @@ private async void Application_Startup(object sender, StartupEventArgs e)
             MainWindow.DataContext = mainVM;
 
             //If language is not recognized, bring up the language selection screen
-            if (!_commons.LanguageCodeMap.ContainsKey(_appSettings.SelectedLanguage))
+            if ((StartupArguments is EmptyArgs || StartupArguments is null) && !_commons.LanguageCodeMap.ContainsKey(_appSettings.SelectedLanguage))
```

Comment:
`StartupArguments` should never be null at this point.

Also it is not just just `EmptyArgs` which should be checked. Both `OpenArgs` and `AdminRestartArgs` should also be taken into account.

Also also refactoring to have two levels of ifs would be clearer
```csharp
if (!_commons.LanguageCodeMap.ContainsKey(_appSettings.SelectedLanguage))
{
  if (StartupArguments is ExportArgs)
  {
    _commons.CurrentLanguage = "English";
  }
  else
  {
    // show welcome
  }
}
else
{
  _commons.CurrentLanguage = _appSettings.SelectedLanguage;;
}
```

Reply:
I changed the control flow so the most common cases are on top.